src/coreclr/src/vm/eventpipe.cpp

```diff
@@ -202,7 +213,8 @@ void EventPipe::EnableViaEnvironmentVariables()
                 EventPipeSessionType::File,
                 EventPipeSerializationFormat::NetTraceV4,
                 true,
-                nullptr
+                nullptr,
+                !s_enableSampleProfilerAtStartup
```

Comment:
I think this needs to just be `false` here.  If you don't configure SampleProfiler as a provider, then `s_enableSampleProfilerAtStartup` will still be `false` and this will call `EventPipe::Enable` with `true` in this parameter.  If I'm reading the code correctly, this will attempt to enable SampleProfiler here which I don't think was the intent.